src/hotspot/cpu/arm/arm.ad

```diff
@@ -869,26 +869,15 @@ uint BoxLockNode::size(PhaseRegAlloc *ra_) const {
 #define R_RTEMP "R_R12"
 void MachUEPNode::format( PhaseRegAlloc *ra_, outputStream *st ) const {
   st->print_cr("\nUEP:");
-  if (UseCompressedClassPointers) {
-    st->print_cr("\tLDR_w " R_RTEMP ",[R_R0 + oopDesc::klass_offset_in_bytes]\t! Inline cache check");
-    st->print_cr("\tdecode_klass " R_RTEMP);
-  } else {
-    st->print_cr("\tLDR   " R_RTEMP ",[R_R0 + oopDesc::klass_offset_in_bytes]\t! Inline cache check");
-  }
+  st->print_cr("\tLDR   " R_RTEMP ",[R_R0 + oopDesc::klass_offset_in_bytes]\t! Inline cache check");
   st->print_cr("\tCMP   " R_RTEMP ",R_R8" );
   st->print   ("\tB.NE  SharedRuntime::handle_ic_miss_stub");
 }
 #endif
 
 void MachUEPNode::emit(CodeBuffer &cbuf, PhaseRegAlloc *ra_) const {
   C2_MacroAssembler _masm(&cbuf);
-  Register iCache  = reg_to_register_object(Matcher::inline_cache_reg_encode());
-  assert(iCache == Ricklass, "should be");
-  Register receiver = R0;
-
-  __ load_klass(Rtemp, receiver);
-  __ cmp(Rtemp, iCache);
-  __ jump(SharedRuntime::get_ic_miss_stub(), relocInfo::runtime_call_type, noreg, ne);
+  __ ic_check(CodeEntryAlignment);
```

Comment:
Do we care about CodeEntryAlignment here if the old code didn't?

Reply:
Perhaps not, but it seems weird not to align. But maybe I should use InteriorEntryAlignment here, so it's symmetric to AArch64.